### ISSUE TITLE
ERNIE: Enhanced Language Representation with Informative Entities

### ISSUE THREAD

**Issue by SOOJEONGKIMM** [SOOJEONGKIMM/Paper_log#11]
https://arxiv.org/pdf/1905.07129.pdf

**Comment by SOOJEONGKIMM** [comment]
Abstract:
Neural language representation models such as BERT pre-trained on large-scale corpora can well capture rich semantic patterns from plain text, and be fine-tuned to consistently improve the performance of various NLP tasks. However, the existing pre-trained language models rarely consider incorporating knowledge graphs (KGs), which can provide rich structured knowledge facts for better language understanding. We argue that informative entities in KGs can enhance language representation with external knowledge. In this paper, we utilize both large-scale textual corpora and KGs to train an enhanced language representation model (ERNIE), which can take full advantage of lexical, syntactic, and knowledge information simultaneously. The experimental results have demonstrated that ERNIE achieves significant improvements on various knowledge-driven tasks, and meanwhile is comparable with the state-of-the-art model BERT on other common NLP tasks. The source code of this paper can be obtained from [this https URL](https://github.com/thunlp/ERNIE)

**Comment by SOOJEONGKIMM** [comment]
Introduction
![image](https://user-images.githubusercontent.com/48557539/192286849-c5d114c8-84d4-41ca-bef9-2e554fd9a880.png)

(1) Structured Knowledge Encoding

encode the graph structure of KGs with knowledge embedding algorithms like TransE and take informative entity embeddings.

(2) Heterogeneous Information Fusion

ERNIE pre-train on large-scale textual corpora and KGs

**Comment by SOOJEONGKIMM** [comment]
Methodology
![image](https://user-images.githubusercontent.com/48557539/192286965-79103d70-9c35-49cd-b5fb-e4f5e39ed632.png)

- Notations

entity to the first token in its named entity phrase.

- Model Architecture

basic lexical and syntactic information.

extra token-oriented

integration of the input of tokens and entities. Information fusion layer takes two kinds of input: one is the token embedding, and the other one is the concatenation of the token embedding and entity embedding. After information fusion, it outputs new token embeddings and entity embeddings for the next layer.

**Comment by SOOJEONGKIMM** [comment]
Knowledgeable Encoder
![image](https://user-images.githubusercontent.com/48557539/192287090-57be47fa-6c70-491f-ae45-0b481b395eb6.png)

**Comment by SOOJEONGKIMM** [comment]
Pre-training for Injecting Knowledge
![image](https://user-images.githubusercontent.com/48557539/192287168-bf283247-521b-4df9-a042-a7ac7ee62592.png)
![image](https://user-images.githubusercontent.com/48557539/192287319-c4c012e5-fabb-4023-9f1e-b875ce77a6a9.png)

**Comment by SOOJEONGKIMM** [comment]
### Fine-tuning for Specific Tasks

relation classification - apply the pooling layer to the final output embeddings of the given entity mentions.

(+) add two mark tokens to highlight entity mentions. [HD] and [TL]

**Comment by SOOJEONGKIMM** [comment]
### Experiments
![image](https://user-images.githubusercontent.com/48557539/192287401-a4ccc48f-73fc-402d-afee-5fecf9305dfc.png)

### Entity Typing
![image](https://user-images.githubusercontent.com/48557539/192287552-bdd47284-e527-4841-9f5a-19cd25e49a9e.png)
BERT and ERNIE make full use of both the unsupervised pre-training and manually annotated training data for better entity typing. 

informative entities help ERNIE predict the labels more precisely.

**Comment by SOOJEONGKIMM** [comment]
### Relation Classification
![image](https://user-images.githubusercontent.com/48557539/192287776-33a2694e-e6c4-4d50-8144-7019d250861d.png)
pre-trained language models can provide more information for relation classification than CNN and RNN. 

especially on FewRel 

**extra knowledge helps the model make full use of small training data**
